spring-test/src/main/java/org/springframework/mock/http/server/reactive/MockServerHttpRequest.java

```diff
@@ -74,16 +74,13 @@
 
 	private final Flux<DataBuffer> body;
 
-
-	private MockServerHttpRequest(@Nullable HttpMethod httpMethod, @Nullable String customHttpMethod,
+	private MockServerHttpRequest(String httpMethodValue,
 			URI uri, @Nullable String contextPath, HttpHeaders headers, MultiValueMap<String, HttpCookie> cookies,
 			@Nullable InetSocketAddress remoteAddress, @Nullable InetSocketAddress localAddress,
 			@Nullable SslInfo sslInfo, Publisher<? extends DataBuffer> body) {
 
 		super(uri, contextPath, headers);
-		Assert.isTrue(httpMethod != null || customHttpMethod != null, "HTTP method must not be null");
-		this.httpMethod = httpMethod;
-		this.customHttpMethod = customHttpMethod;
```

Comment:
We still can assert the httpMethodValue is not null.

Reply:
I thought that all the checks are done in the Builder and the constructor is private. 

All classes extending the constructor must call super(), thus made sure there is no path able to initialize a null HttpMethod. 

But if you want I can surely add it

Reply:
@rstoyanchev I added some easy tests, testing all possible build paths. 

Should I anyways add the assertion here?